### PR TITLE
DSD-708: Fixing nested footer elements in TemplateAppContainer

### PR DESCRIPTION
**Pull request by EdwinGuzman**
Fixes JIRA ticket [DSD-708](https://jira.nypl.org/browse/DSD-708)

## This PR does the following:

- This fixes an issue when a custom component meant to render a `<footer>` element is passed to the `TemplateAppContainer` or `TemplateFooter` components. For the new `renderFooterElement` prop, when it is set to false, a default `footer` element won't be rendered. This fixes the nested HTML `<footer>` elements.

## How has this been tested?

Storybook and tests.

## Accessibility concerns or updates

- Not anymore. It's not a syntax issue but a semantic issue to have nested footer elements.

### Checklist:

- [ ] I have updated the Storybook documentation accordingly.
- [ ] I have added relevant accessibility documentation for this pull request.
- [ ] All new and existing tests passed.

### Front End Review:

- [ ] View [the example in Storybook]()

### CODE REVIEW

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr841-88nea9geux34jyun5gy92lyaezq9qnlo.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/61ddf2885f46ed13d82415ec

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr841-88nea9geux34jyun5gy92lyaezq9qnlo.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/61ddf2885f46ed13d82415ec

File: src/components/Template/Template.tsx [hunk 1]
```diff
@@ -152,10 +158,18 @@ const TemplateContentSidebar = (
...
+ * custom component, set `renderFooterElement` to `false`.
  */
-const TemplateFooter = (props: React.PropsWithChildren<TemplateProps>) => {
-  return <Box as="footer">{props.children}</Box>;
+const TemplateFooter = ({
+  children,
+  renderFooterElement = true,
+}: React.PropsWithChildren<TemplateFooterProps>) => {
+  return renderFooterElement ? (
+    <Box as="footer">{children}</Box>
```
Comment (bigfishdesign13) [inline src/components/Template/Template.tsx]: Can you add a warning message if a `footer` element is passed in as a child in this scenario?

**Review by bigfishdesign13 — CHANGES_REQUESTED** [review]
I asked to one addition.  Let me know if you think it is unnecessary.

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr841-88nea9geux34jyun5gy92lyaezq9qnlo.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/61ddf2885f46ed13d82415ec

**Review by bigfishdesign13 — APPROVED** [review]
Looks good.

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr841-88nea9geux34jyun5gy92lyaezq9qnlo.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/61ddf2885f46ed13d82415ec

**Pull request merged**